src/TMGToolbox/inputOutput/importMatrixFromCSVThirdNormalized.py

```diff
@@ -15,149 +15,273 @@
-            if value != 0.0:
-                if vehicleEID == 'transit':
-                    originEID = f"ped_entrance_centroid_{line[0]}"
-                    destinationEID = f"ped_exit_centroid_{line[1]}"
-                    origin = catalog.findObjectByExternalId(originEID, entranceCentroidType)
-                    destination = catalog.findObjectByExternalId(destinationEID, exitCentroidType)
-                else:
-                    originEID = f"centroid_{line[0]}"
-                    destinationEID = f"centroid_{line[1]}"
-                    origin = catalog.findObjectByExternalId(originEID, sectionType)
-                    destination = catalog.findObjectByExternalId(destinationEID, sectionType)
-                if origin is None:
-                    raise Exception(f"The specified centroid '{originEID}' does not exist")
-                if destination is None:
-                    raise Exception(f"The specified centroid '{destinationEID}' does not exist")
-                matrix.setTrips(origin, destination, value)
+        for line in reader:
+            if line[1] == argv[4] and len(line)>=6:
+                xtmf_parameters['matrix_id'] = line[1]
+                xtmf_parameters['centroid_configuration'] = line[2]
+                xtmf_parameters['vehicle_type'] = line[3]
+                xtmf_parameters['initial_time'] = line[4]
+                xtmf_parameters['duration_time'] = line[5]
+
+    fileLocation = str(xtmf_parameters['matrix_location'])
+    thirdNormalized = bool(xtmf_parameters['third_normalized'])
+    header = bool(xtmf_parameters['includes_header'])
+    matrixId = str(xtmf_parameters['matrix_id'])
+    centroidConfigurationId = str(xtmf_parameters['centroid_configuration'])
+    vehicleEID = str(xtmf_parameters['vehicle_type'])
+    initialTime = str(xtmf_parameters['initial_time'])
+    durationTime = str(xtmf_parameters['duration_time'])
+
+    # Start a console
+    console = ANGConsole()
+    model = None
+    # Load a network
+    if console.open(argv[1]): 
+        model = console.getModel()
+        print("open network")
+    else:
+        console.getLog().addError("Cannot load the network")
+        raise Exception("cannot load network")
+
+    catalog = model.getCatalog()
+    #find the centroid configuration
+    sectionType = model.getType("GKCentroidConfiguration")
+    centroidConfiguration = catalog.findObjectByExternalId(centroidConfigurationId, sectionType)
+    if centroidConfiguration is None:
+        raise Exception(f"The specified centroid configuration '{centroidConfigurationId}' does not exist")
+    #check if matrix with given id exists and delete it if it does
+    if catalog.findObjectByExternalId(matrixId) != None:
+        while catalog.findObjectByExternalId(matrixId) != None:
+            obj = catalog.findObjectByExternalId(matrixId)
+            cmd = obj.getDelCmd()
+            model.getCommander().addCommand(cmd)
+            model.getCommander().addCommand(None)
```

Comment:
Are we sure that you need to add a null command to the buffer?  This feels wrong to me.

Reply:
I didn't write this code it's Spencer's code we would have to ask Spencer for clarification on this